weblab/accounts/tests/test_views.py

```diff
@@ -1,5 +1,62 @@
+import pytest
+
+from accounts.models import User
+from core import recipes
+from datasets.models import Dataset
+
 
 def test_my_account_view_requires_login(client):
     response = client.get('/accounts/myaccount/')
     assert response.status_code == 302
     assert '/login/' in response.url
+
+
+@pytest.mark.django_db
+def test_user_can_delete_own_account(client, logged_in_user, my_dataset):
+    recipes.model.make(author=logged_in_user)
+    dataset = my_dataset
+    user_directory_repo = logged_in_user.get_storage_dir('repo')
+    user_directory_dataset = logged_in_user.get_storage_dir('dataset')
+    assert Dataset.objects.filter(pk=dataset.pk).exists()
+
+    assert user_directory_repo.is_dir()
+    assert user_directory_dataset.is_dir()
+
+    response = client.post(
+        '/accounts/%d/delete/' % logged_in_user.pk,
+    )
+
+    assert response.status_code == 302
+    assert not User.objects.filter(pk=logged_in_user.pk).exists()
```

Comment:
We could check where this redirect is to; I think it should be `assert response.url == reverse('home')`?